djpaddle/templates/djpaddle_post_checkout.html

```diff
@@ -19,27 +14,49 @@
         passthrough: data.checkout.passthrough,
         email: email,
         created_at: data.checkout.created_at,
+        redirect_url: data.checkout.redirect_url,
     };
     var encodedString = buildFormData(postData);
 
+    var xhr = new XMLHttpRequest();
+    xhr.open("POST", "{% url 'djpaddle:post_checkout_api' %}?next={{ djpaddle_checkout_success_redirect }}");
+    xhr.setRequestHeader("Content-Type", "application/x-www-form-urlencoded");
+    xhr.setRequestHeader("X-CSRFToken", csrftoken);
+
+    xhr.onreadystatechange = function(){
+      // Process the server response here.
+        if (xhr.readyState === XMLHttpRequest.DONE) {
+            console.log(xhr.status);
+            if (xhr.status === 200) {
+                var jsonResponse = JSON.parse(xhr.responseText);
+                if (jsonResponse.hasOwnProperty("redirect_url")) {
+                    window.location.href = jsonResponse["redirect_url"];
+                }
+            } else if (xhr.status === 204) {
+                console.log("Checkout complete with ID: " + data.checkout.id)
+            } else {
+                alert("There was a problem with the request.");
```

Comment:
@pyepye this should rather redirect to an error-page

Reply:
I'm not sure it's a good idea to redirect to any error page as it's not the Paddle checkout that has failed. It's just the dj-paddle post checkout request which isn't _that_ important as a webhook should come along soon and ensure we know the checkout was a success.